### PR TITLE
feat: Design system version 2

### PR DESCRIPTION
**Pull request by albinAppsmith**
## Description

This PR contains basic set up for the design system v2. 

Fixes [#19651](https://github.com/appsmithorg/appsmith/issues/19651)

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Tested in local

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by yaldram — APPROVED** [review]

**Review by tanvibhakta — APPROVED** [review]

File: packages/design-system-v2/package.json [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        }
+      }
+    ]
+  },
+  "resolutions": {
+    "styled-components": "^5"
+  },
+  "dependencies": {
+    "@react-aria/button": "^3.6.4",
+    "styled-components": "^5.3.6"
```
Comment (yaldram) [inline packages/design-system-v2/package.json:94]: styled-components should also be in the peer dependencies.

**Review by yaldram — CHANGES_REQUESTED** [review]

**Review by tanvibhakta — APPROVED** [review]

File: packages/design-system-v2/package.json [hunk 2]
```diff
@@ -1,77 +1,61 @@
...
-    "@babel/preset-typescript": "^7.17.12",
+    "@babel/core": "^7.20.12",
     "@react-aria/focus": "^3.10.1",
     "@react-aria/radio": "^3.4.2",
     "@react-types/button": "^3.7.0",
-    "@rollup/plugin-commonjs": "^24.0.0",
-    "@rollup/plugin-image": "^3.0.1",
-    "@rollup/plugin-terser": "^0.3.0",
-    "@rollup/plugin-url": "^8.0.1",
+    "@rollup/plugin-typescript": "^11.0.0",
```
Comment (tanvibhakta) [inline packages/design-system-v2/package.json:23]: If we're removing rollup why do we still need the plugin for it?

File: packages/design-system-v2/src/box.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
```
Comment (tanvibhakta) [inline packages/design-system-v2/src/box.stories.tsx:1]: What's this?

File: packages/design-system-v2/vite.config.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import { defineConfig } from "vite";
+import typescript from "@rollup/plugin-typescript";
```
Comment (tanvibhakta) [inline packages/design-system-v2/vite.config.ts:2]: interesting

File: packages/design-system-v2/package.json (hunk 2 above)
Comment (albinAppsmith) [reply packages/design-system-v2/package.json:23]: We need this to support typescript in Vite

File: packages/design-system-v2/src/box.stories.tsx (hunk 3 above)
Comment (albinAppsmith) [reply packages/design-system-v2/src/box.stories.tsx:1]: This I need to test storybook hosting. Will remove afterwards

File: packages/design-system-v2/vite.config.ts (hunk 4 above)
Comment (albinAppsmith) [reply packages/design-system-v2/vite.config.ts:2]: Required.

File: packages/design-system-old/.editorconfig [hunk 5]
```diff
@@ -0,0 +1,12 @@
+root = true
```
Comment (tanvibhakta) [inline packages/design-system-old/.editorconfig:1]: Do you want to .gitignore this?

Comment (albinAppsmith) [reply packages/design-system-old/.editorconfig:1]: No, this is required file.

File: packages/design-system/package.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "@appsmithorg/design-system",
-  "version": "1.0.45",
-  "description": "This is the package for the Appsmith design system components",
+  "name": "@appsmithorg/design-system-v2",
+  "version": "2.0.0-alpha.0",
+  "description": "",
```
Comment (tanvibhakta) [inline packages/design-system/package.json:4]: "description": "This is the package for the design system that powers the Appsmith platform",

**Review by yaldram — APPROVED** [review]

File: packages/design-system/plop-templates/stories.tsx.hbs [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { ComponentMeta, ComponentStory } from "@storybook/react";
+
+import { {{capitalize name}} } from "./{{capitalize name}}";
+
+export default {
+  title: "Design System V2/{{capitalize name}}",
```
Comment (tanvibhakta) [inline packages/design-system/plop-templates/stories.tsx.hbs:7]:
```suggestion
title: "Design System/{{capitalize name}}",
```

File: packages/design-system/package.json [hunk 8]
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "@appsmithorg/design-system",
-  "version": "1.0.45",
-  "description": "This is the package for the Appsmith design system components",
+  "name": "@appsmithorg/design-system-v2",
```
[the remark this answers is not in the document]
Comment (albinAppsmith) [reply packages/design-system/package.json]: Updated this

**Review by tanvibhakta — APPROVED** [review]

**Pull request merged**